tensorflow/python/ops/custom_gradient.py

```diff
@@ -482,26 +482,42 @@ def recompute_grad(f):
   def inner(*args, **kwargs):
     """Inner function closure for calculating gradients."""
     current_var_scope = variable_scope.get_variable_scope()
-
-    result = f(*args, **kwargs)
-
+    with tape_lib.stop_recording():
+      result = f(*args, **kwargs)
+    @custom_gradient
     def grad(*dresult, **grad_kwargs):
-      """Gradient function calculation for inner function."""
-      variables = grad_kwargs.get("variables")
-      with backprop.GradientTape() as t:
-        id_args = [gen_array_ops.identity(x) for x in args]
-        t.watch(id_args)
+      """Nested custom gradient function for computing grads in reverse and forward mode autodiff."""
+
+      def grad_eval():
```

Comment:
Any reason to have this inner nested function? I think you can take it out and save some indentation

Reply:
No reason. Removed it.